src/cargo/ops/registry.rs

```diff
@@ -388,6 +388,13 @@ pub fn configure_http_handle(config: &Config, handle: &mut Easy) -> CargoResult<
     } else {
         handle.useragent(&version().to_string())?;
     }
+
+    let noproxy = env::var("NO_PROXY")
```

Comment:
Should this be `NO_PROXY` or `no_proxy`?

Reply:
Oh this is just respecting curl's existing convention with the env var `NO_PROXY`. Although did you mean the local variable should be called `no_proxy` (aka add an underscore?)

Reply:
The environment variables in #6114 used `no_proxy` rather than `NO_PROXY`. IIRC curl has a couple of other mysteriously lower case variables as well?

Reply:
Sure we can add support for both, curl does indeed support both